src/neuroconv/datainterfaces/behavior/video/internalvideointerface.py

```diff
@@ -0,0 +1,361 @@
+        module_description: Optional[str] = None,
+    ):
+        """
+        Convert the video data files to :py:class:`~pynwb.image.ImageSeries` and write them in the
+        :py:class:`~pynwb.file.NWBFile`. Data is written in the :py:class:`~pynwb.image.ImageSeries` container as
+        RGB. [times, x, y, 3-RGB].
+
+        Parameters
+        ----------
+        nwbfile : NWBFile, optional
+            nwb file to which the recording information is to be added
+        metadata : dict, optional
+            Dictionary of metadata information such as names and description of each video.
+            Metadata should be passed for each video file passed in the file_paths.
+            If storing as 'external mode', then provide duplicate metadata for video files that go in the
+            same :py:class:`~pynwb.image.ImageSeries` container.
+            Should be organized as follows::
+
+                metadata = dict(
+                    Behavior=dict(
+                        Video=dict(
+                            InternalVideo=dict(
+                                description="Description of the video..",
+                                unit="Frames",
+                                ...,
+                            ),
+                        )
+                    )
+                )
+
+            and may contain most keywords normally accepted by an ImageSeries
+            (https://pynwb.readthedocs.io/en/stable/pynwb.image.html#pynwb.image.ImageSeries).
+            Each dictionary in the list corresponds to a single VideoInterface and ImageSeries.
+        stub_test : bool, default: False
+            If ``True``, truncates the write operation for fast testing.
+        chunk_data : bool, default: True
+            If True, uses a DataChunkIterator to read and write the video, reducing overhead RAM usage at the cost of
+            reduced conversion speed (compared to loading video entirely into RAM as an array). This will also force to
+            True, even if manually set to False, whenever the video file size exceeds available system RAM by a factor
+            of 70 (from compression experiments). Based on experiments for a ~30 FPS system of ~400 x ~600 color
+            frames, the equivalent uncompressed RAM usage is around 2GB per minute of video. The default is True.
+        parent_container: {'acquisition', 'processing/behavior'}
+            The container where the ImageSeries is added, default is nwbfile.acquisition.
+            When 'processing/behavior' is chosen, the ImageSeries is added to nwbfile.processing['behavior'].
+        module_description: str, optional
+            If parent_container is 'processing/behavior', and the module does not exist,
+            it will be created with this description. The default description is the same as used by the
+            conversion_tools.get_module function.
+        """
+        if not chunk_data:
```

Comment:
Ah, I see, yeah, I think it should be like the other interfaces. There is a way to not use iterators.

Reply:
Ok, I updated accordingly.